libraries/chain/block_state_legacy.cpp

```diff
@@ -75,7 +75,7 @@ namespace eosio::chain {
    ,block( std::move(b) )
    ,_pub_keys_recovered( true ) // called by produce_block so signature recovery of trxs must have been done
    ,_cached_trxs( std::move(trx_metas) )
-   ,action_receipt_digests_savanna( std::move(action_receipt_digests_savanna) )
+   ,action_mroot_savanna( action_receipt_digests_savanna ? std::optional<digest_type>(calculate_merkle(*action_receipt_digests_savanna)) : std::nullopt )
```

Comment:
let's make the `action_receipt_digests_savanna` parameter to the constructor be a `const&`, we don't need it ro be a rvalue anymore.